src/voprf.rs

```diff
@@ -908,6 +833,46 @@ where
 // =============== //
 /////////////////////
 
+#[allow(clippy::type_complexity)]
+pub(crate) fn derive_keypair<CS: CipherSuite>(
+    seed: &[u8],
+    info: &[u8],
+    mode: Mode,
+) -> Result<(<CS::Group as Group>::Scalar, <CS::Group as Group>::Elem), Error>
+where
+    <CS::Hash as OutputSizeUser>::OutputSize:
+        IsLess<U256> + IsLessOrEqual<<CS::Hash as BlockSizeUser>::BlockSize>,
+{
+    let context_string = create_context_string::<CS>(mode);
+    let dst = GenericArray::from(STR_DERIVE_KEYPAIR).concat(context_string);
+
+    // deriveInput = seed || I2OSP(len(info), 2) || info
+    let info_len = i2osp_2(info.len()).map_err(|_| Error::Metadata)?;
+
+    let mut counter: usize = 0;
+
+    loop {
+        if counter > 255 {
+            break Err(Error::DeriveKeyPair);
+        }
+
+        // skS = G.HashToScalar(deriveInput || I2OSP(counter, 1), DST = "DeriveKeyPair"
+        // || contextString)
+        let counter_i2osp = i2osp_1(counter).map_err(|_| Error::DeriveKeyPair)?;
```

Comment:
If we let `counter` be a `u8`, we can simply use `.to_be_bytes()` here, which also can't fail.